.gitlab/ci/on-push.yml

```diff
@@ -37,6 +37,42 @@ trigger-private-build:
   - ./Documentation/commonServerDocs.sh
   - section_end "Create Release Notes and Common Server Documentation"
 
+stop-running-pipelines:
+  tags:
+    - gke
+  stage: unittests-and-validations
+  needs: []
+  variables:
+    master_branch_name: master
+  rules:
+    - if: '$CI_PIPELINE_SOURCE =~ /^(push)$/ && $CI_COMMIT_BRANCH != $master_branch_name'
+  script:
+    - section_start "Stop running piplined on current branch"
+    - |
+      if [[ $CI_COMMIT_BRANCH = "master" ]]; then # for testing purpose only will be removed
+        echo "Skipping, Should not run on master branch."
+      else
+        echo $CI_PIPELINE_SOURCE
+        echo $CI_COMMIT_BRANCH
+        echo "curling to gitlab"
+        RESP=$(curl -v -H "Content-Type: application/json" -H "PRIVATE-TOKEN: $GITLAB_CANCEL_TOKEN" https://code.pan.run/api/v4/projects/2596/pipelines\?ref\=$CI_COMMIT_BRANCH)
+        echo $RESP
+        if [ "$RESP" != "[]" ]; then
+            PIPELINES=$(echo "${RESP}" | jq -c '.[]')
+            for pipeline in ${PIPELINES}
+            do
+                status=$(echo $pipeline | jq -r ".status")
+                id=$(echo $pipeline | jq -r ".id")
+                if [ "$status" = "running" ]; then
+                    echo "Found running pipeline with id $id, and status $status. Stoping it."
```

Comment:
You are canceling all existing builds that run on this branch.
But if I want to trigger nightly and upload on this branch, after I will push one more time this step will cancel my triggered nightlies and test-uploads.

You should check the source as well: `"source": "trigger"` is for nightly and upload, and `"source": "push"` is for the regular push.

Please add one more condition: cancel build `if status is "running" and source is "push"`.